plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -63,36 +75,60 @@ export const EntityPicker = (props: EntityPickerProps) => {
   const defaultKind = uiSchema['ui:options']?.defaultKind;
   const defaultNamespace =
     uiSchema['ui:options']?.defaultNamespace || undefined;
+  const allowArbitraryValues =
+    uiSchema['ui:options']?.allowArbitraryValues ?? true;
+  const optionLabelSchema =
+    allowArbitraryValues !== true
+      ? uiSchema['ui:options']?.optionLabelSchema
+      : undefined;
+  const additionalFields = getOptionLabelFields(optionLabelSchema);
 
   const catalogApi = useApi(catalogApiRef);
 
   const { value: entities, loading } = useAsync(async () => {
-    const fields = ['metadata.name', 'metadata.namespace', 'kind'];
+    const fields = [
+      ...new Set([
+        'metadata.name',
+        'metadata.namespace',
+        'kind',
+        ...additionalFields,
+      ]),
+    ];
     const { items } = await catalogApi.getEntities(
       catalogFilter
         ? { filter: catalogFilter, fields }
         : { filter: undefined, fields },
     );
     return items;
   });
-  const allowArbitraryValues =
-    uiSchema['ui:options']?.allowArbitraryValues ?? true;
 
-  const getLabel = useCallback(
-    (ref: string) => {
-      try {
-        return humanizeEntityRef(
-          parseEntityRef(ref, { defaultKind, defaultNamespace }),
-          {
-            defaultKind,
-            defaultNamespace,
-          },
-        );
-      } catch (err) {
-        return ref;
+  const getOptionLabel = useCallback(
```

Comment:
just to flag that `useCallback` can be removed as `additionalFields` is not stable